src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -336,7 +336,13 @@ ExpressionEvaluator::_evaluate_in_expression<ExpressionEvaluator::Bool>(const In
     });
 
   } else {
-    Fail("Unsupported ExpressionType used in InExpression");
+    /**
+     * `<expression> IN <anything>` is not legal SQL, but on expression level we have to support it, since `<anything>`
+     * might be a column holding the result of a subselect.
+     * To accomplish this, we simply rewrite the expression to `<expression> IN LIST(<anything>)`
+     */
+
+    return _evaluate_in_expression<ExpressionEvaluator::Bool>(*in_(in_expression.value(), list_(in_expression.set())));
```

Comment:
I don't understand this yet. Why did this become necessary now? What does this mean for this branch:

https://github.com/hyrise/hyrise/blob/c83d85b533688de22defbc56b56db1cbb69ac65a/src/lib/expression/evaluation/expression_evaluator.cpp#L284

Also, does this mean that you now build that long `OR` list instead of going through the result?

Reply:
To clarify: I understand that PQPSelects are now rewritten to PQPColumns, correct? But wouldn't we still want to handle them the same way?